lib/logstash/inputs/file.rb

```diff
@@ -181,27 +182,47 @@ def register
     if @start_position == "beginning"
       @tail_config[:start_new_files_at] = :beginning
     end
+
+    if (@run_with_identity = @codec.class.config_name == 'multiline')
+      @codec = LogStash::Codecs::IdentityMapCodec.new(@codec)
+    end
```

Comment:
Would it be easier to just go through the same `encode` code every time regardless of the need to an `identity_stream or not?` I though this was the initial goal of implementing a similar interface to not have to worry about codec that might need it and keeping them dumb.

 But I might be wrong here, since this would reduce the memory foot print of the application.